### ISSUE TITLE
Bug report

### ISSUE THREAD

**Issue by akds-singh** [localsend/localsend#1591]
### Describe the bug

Not able to save .mkv file to Ipad from Android or windows

### To reproduce

1. choose a .mkv file either Android or window
2. select iPad to send a .mkv file from Android or winndow
3. let's complete the uploading , at finished point erroe will apear
5. [500] Could not save file

### Expected behavior

file need to save in Ipad from android/window

### Screenshots

![image](https://github.com/user-attachments/assets/12dfaceb-7aee-44f3-a8e0-74827a919c07)

**Comment by Tienisto** [comment]
You need to provide the error message on the receiver. This error message is not very helpful

**Comment by akds-singh** [comment]
Actually, there is not displaying or pop-up any type of error in receipt side(ipad) . After completion of the uploading from sender side( windows here) , above error is displaying ( error: 500) but there is no error displayed recepeint side.

**Comment by Tienisto** [comment]
Okay, this is weird. Can you disable "Save media to gallery / photos"?

**Comment by akds-singh** [comment]
Yes, by disabling 'save media to gallery ' it worked and file saved in default folder in recipient side.

Thanks 🙏 , for building such a great application.

**Comment by Tienisto** [comment]
Can you please provide the app version? I cannot reproduce this error on my device

**Comment by km5464x** [comment]
> Can you please provide the app version? I cannot reproduce this error on my device

just came across this issue, the app version is 1.15.4 (54).

Any updates please do lmk 💪🏾

**Comment by Hooey** [comment]
I also get this [500] error: Could not save file. Check destination for more info (paraphrased)
The receiving device doesn't have more info. I disabled 'save media to gallery' and that made no difference. I don't know where to find the app version. It's not under "About LocalSend" that I could see.

**Comment by km5464x** [comment]
From my troubleshooting. It seemed to be the file format. I attempted to transfer a video. I forgot the specific format that made it bug, but as soon as I changed it to MP4, it works smoothly.

**Comment by Hooey** [comment]
In my case this time it was a .cue file. I didn't try changing the format; didn't know if I could. I guess I can just try renaming it.

**Comment by Loyalsoldier** [comment]
https://github.com/localsend/localsend/issues/1309#issuecomment-2568680226

**Comment by theLittlePrinceAD** [comment]
Receiving error:
[RhttpStatusCodeException] Status code: 500. URL: https://192.168.100.4:53317/api/localsend/v2/upload

but files are copied and are OK.

**Comment by wonderhj** [comment]
我在传DWG格式 的文件的时候出现这个问题，传其他格式又没有问题。不知道啥原因。

**Comment by wonderhj** [comment]
文件名里有个下划线 _   是不是这个的原因？

**Comment by morefeee** [comment]
Transferred MKV files from PC(win11 23H2) to iPad(iPadOS 18.4.1), encountered an error upon completion.
[RhttpStatusCodeException] Status code: 500. URL: https://192.168.0.113:53317/api/localsend/v2/upload

**Comment by jinggqu** [comment]
> I also get this [500] error: Could not save file. Check destination for more info (paraphrased) The receiving device doesn't have more info. I disabled 'save media to gallery' and that made no difference. I don't know where to find the app version. It's not under "About LocalSend" that I could see.

Can reproduce.

The possible reason is that the gallery on iOS only supports common file formats (jpg, png, mp4, etc.), excluding mkv, etc. 

This problem can be resolved by **turning off** the `save to gallery` option, which align with the above conjecture.

**Comment by theLittlePrinceAD** [comment]
> > I also get this [500] error: Could not save file. Check destination for more info (paraphrased) The receiving device doesn't have more info. I disabled 'save media to gallery' and that made no difference. I don't know where to find the app version. It's not under "About LocalSend" that I could see.
> 
> Can reproduce.
> 
> The possible reason is that the gallery on iOS only supports common file formats (jpg, png, mp4, etc.), excluding mkv, etc.
> 
> This problem can be resolved by **turning off** the `save to gallery` option, which align with the above conjecture.

Me being on Android and Save to Gallery is turned off.

**Comment by OlegShirochenkov** [comment]
I've encountered a case mentioned in a comments: RhttpStatusCodeException when transferring a particular file from iOS to Windows. The reason was a restricted symbols in a filename: colon (:)

**Comment by zhangyangeng** [comment]
I'm going to give you a more partial solution, which is not suitable for most people, but only for people who are in the same situation as me.
When transferring data from macOS/Windows to Android, which is rooted, an error message "[RhttpStatusCodeException] Status code: 500. URL: https://xxx:53317/api/localsend/v2/upload" will appear. However, there is no problem transferring data between macOS and Windows. I suspect it is a problem with the phone. When I unchecked the "Enable storage isolation" switch in the "Storage Isolation" software, I can save the data normally. I suspect it may be related to storage. Can anyone who knows about it help explain it to me?

![Image](https://github.com/user-attachments/assets/a34b08f9-8084-40ab-b36e-fa37a8799d7a)

**Comment by notGabo** [comment]
> Okay, this is weird. Can you disable "Save media to gallery / photos"?

almost 2026, this worked for me on iphone 17

**Comment by unixcs** [comment]
我也遇到了这个问题，我尝试。清除安卓端 APP 数据。然后再次尝试发送文件。成功解决了。
